cmd/abci-cli/abci-cli.go

```diff
@@ -196,6 +202,45 @@ func persistentArgs(line []byte) []string {
 
 //--------------------------------------------------------------------------------
 
+func cmdTest(app *cli.App, c *cli.Context) error {
+	fmt.Println("Running tests")
+
+	var err error
+
+	err = servertest.InitChain(client)
+	fmt.Println("")
+	err = servertest.SetOption(client, "serial", "on")
+	fmt.Println("")
+	err = servertest.Commit(client, nil)
+	fmt.Println("")
+	err = servertest.DeliverTx(client, []byte("abc"), types.CodeType_BadNonce, nil)
+	fmt.Println("")
+	err = servertest.Commit(client, nil)
+	fmt.Println("")
+	err = servertest.DeliverTx(client, []byte{0x00}, types.CodeType_OK, nil)
+	fmt.Println("")
+	err = servertest.Commit(client, []byte{0, 0, 0, 0, 0, 0, 0, 1})
+	fmt.Println("")
+	err = servertest.DeliverTx(client, []byte{0x00}, types.CodeType_BadNonce, nil)
+	fmt.Println("")
+	err = servertest.DeliverTx(client, []byte{0x01}, types.CodeType_OK, nil)
+	fmt.Println("")
+	err = servertest.DeliverTx(client, []byte{0x00, 0x02}, types.CodeType_OK, nil)
+	fmt.Println("")
+	err = servertest.DeliverTx(client, []byte{0x00, 0x03}, types.CodeType_OK, nil)
+	fmt.Println("")
+	err = servertest.DeliverTx(client, []byte{0x00, 0x00, 0x04}, types.CodeType_OK, nil)
+	fmt.Println("")
+	err = servertest.DeliverTx(client, []byte{0x00, 0x00, 0x06}, types.CodeType_BadNonce, nil)
+	fmt.Println("")
+	err = servertest.Commit(client, []byte{0, 0, 0, 0, 0, 0, 0, 5})
+
+	if err != nil {
+		return errors.New("Some checks didn't pass, please use the cli to see the exact failures.")
```

Comment:
huh? this command is a cli command. what do you mean by `use the cli`?

Reply:
i think he means "debug manually", like if in running `abci-cli test` there's a failure at one of the lines, use abci-cli deliver_tx` or whatever to debug. Instead, printing the actual `err` message on this line would show the exact failures, would it not?

Reply:
I’ll change “use the cli” to “inspect stdout”.